### PR TITLE
XmlWebViewInsideScrollView Detector

### PR DESCRIPTION
**Pull request by vickyramachandra**
This PR adds a lint check to make sure that Scrollview which has WebView as its child sets fillViewport to true

  - [x] Add `XmlWebViewInsideScrollViewDetector`

  - [x] Add Unit tests

### CODE REVIEW

File: lint-checks-android/src/main/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetector.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    parentScrollView?.attributes?.let { attrs ->
+      if (attrs.getNamedItem(ATTR_ANDROID_VIEWPORT) == null ||
+          attrs.getNamedItem(ATTR_ANDROID_VIEWPORT).nodeValue != "true") {
+        val replaceFix = LintFix.create()
+            .set()
+            .attribute(ATTR_FILL_VIEWPORT)
+            .value("true")
+            .build()
+        context.report(ISSUE,
+            context.getElementLocation(parentScrollView), LINT_ERROR_MESSAGE,
```
Comment (ShaishavGandhi) [inline lint-checks-android/src/main/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetector.kt]: Nit: Move the LINT_ERROR_MESSAGE to new line for consistency?

File: lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+import com.uber.lintchecks.base.test.LintTestBase
+import org.junit.Test
+
+class XmlWebViewInsideScrollViewDetectorTest : LintTestBase() {
+  @Test
+  fun testDetector_scrollViewWithoutFillViewPort_shouldFail() {
+    TestLintTask.lint()
+        .files(xmlSource("res/layout/webview_inside_scrollview_wo_fill_viewport.xml",
+            """<?xml version="1.0" encoding="utf-8"?>
+                            <ScrollView
```
Comment (ShaishavGandhi) [inline lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt]: Nit: These are indented weird. <ScrollView> and <?xml version> should be on the same indent level

File: lint-checks-android/src/main/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetector.kt (hunk 1 above)
Comment (vickyramachandra) [reply lint-checks-android/src/main/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetector.kt]: sure, will do

File: lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt (hunk 2 above)
Comment (vickyramachandra) [reply lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt]: I see, let me format them

Comment (vickyramachandra) [reply lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt]: @ShaishavGandhi i ran a reformat on the file. Still this indent remained

Comment (vickyramachandra) [reply lint-checks-android/src/test/java/com/uber/lintchecks/android/XmlWebViewInsideScrollViewDetectorTest.kt]: Done. Manually formatted

**Comment by ShaishavGandhi** [comment]
Can you run spotless again?

**Comment by vickyramachandra** [comment]
> Can you run spotless again? It's failing the build

Yeah fixed

**Pull request merged**